code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -123,13 +123,22 @@
 	user.SetWeakened(0)
 	user.incorporeal_move = 1
 	user.alpha = 0
+	user.ExtinguishMob()
+	var/turf/T = get_turf(user)
+	user.forceMove(T) //to properly move the mob out of a potential container
 	if(user.buckled)
 		user.buckled.unbuckle_mob()
+	if(user.pulledby)
+		user.pulledby.stop_pulling()
+	user.stop_pulling()
+	if(user.buckled_mob)
+		user.unbuckle_mob(force=1)
```

Comment:
slime buckles itself on shadowling, shadowling use shadow walk, the slime enjoys the ride...